doc-deploy-dev/action.yml

```diff
@@ -20,123 +30,145 @@ inputs:
+        repository: ${{ env.REPOSITORY }}
+        ref: ${{ inputs.branch }}
+        token: ${{ inputs.token }}
 
-    - name: "Create a fresh development folder"
-      shell: bash
-      run: |
-        rm -rf dev/ && mkdir dev/
+    # Downlaod the documentation artifact from the current workflow. If the
+    # artifact contains a compressed file, decompress it. Display the structure
+    # of the 'version/dev' directory at the end of the process to verify the
+    # layout of the folder is the right one.
 
-    - name: "Download the HTML documentation artifact from desired workflow"
-      uses: dawidd6/action-download-artifact@v2
-      if: inputs.workflow != ''
-      with:
-        github_token: ${{ inputs.token }}
-        name: ${{ inputs.doc-artifact-name }}
-        path: dev
-        workflow: ${{ inputs.workflow }}
-        workflow_conclusion: ${{ inputs.workflow-conclusion }}
+    - name: "Clean destination folder"
+      shell: bash
+      run:
+        rm -rf version/dev && mkdir -p version/dev
 
-    - name: "Download the HTML documentation artifact from current workflow"
+    - name: "Download the development documentation artifact"
       uses: actions/download-artifact@v3
-      if: inputs.workflow == ''
       with:
         name: ${{ inputs.doc-artifact-name }}
-        path: dev
+        path: version/dev
 
-    - name: "Decompress artifact content if required"
+    - name: "Decompress artifact content"
       shell: bash
-      if: inputs.decompress_artifact == 'true'
-      working-directory: dev
+      if: inputs.decompress-artifact == 'true'
       run: |
-        compressed_artifact=$(ls .)
-        ouch decompress $compressed_artifact
-        rm $compressed_artifact
+        sudo apt-get install cargo && cargo install ouch && ouch --version
+        cd version/dev && compressed_artifact=$(ls .)
+        ouch decompress $compressed_artifact && rm $compressed_artifact
 
-    - name: "Display structure of downloaded files"
+    - name: "Display structure of version/dev"
       shell: bash
       run: |
-        ls -R
+        ls -R version/dev
 
-    - name: "Include CNAME and nojekyll files expected by GitHub pages"
+    # Create the 'versions.json' file if it does not exist. Note that this step
+    # is not required in the 'doc-deploy-stable' action. The reason is that
+    # stable releases are born after a development version exists. This forces
+    # users to use this action before using the stable deploy one.
+
+    - name: "Create the 'versions.json' file if not present"
       shell: bash
       run: |
-        touch CNAME .nojekyll
-        echo ${{ inputs.cname }} > CNAME
+        if ! [[ -f "versions.json" ]];
+        then
+            sudo apt-get install moreutils jq
+            echo '[]' > versions.json
+            url_dev="https://${{ inputs.cname }}/version/dev/"
+            jq --arg url $url_dev '. += [{"name": "dev", "version": "dev", "url": $url}]' versions.json | sponge versions.json
```

Comment:
This is definitely going to my CLI utils list

Reply:
It turned out to be a key tool when migrating the whole tool to shell script.